apps/miiverse-api/src/services/internal/routes/posts.ts

```diff
@@ -0,0 +1,17 @@
+import express from 'express';
+import { getPostByID } from '@/database';
+import { errors } from '@/services/internal/errors';
+import { handle } from '@/services/internal/utils';
+import { guards } from '@/services/internal/middleware/guards';
+import { mapPost } from '@/services/internal/contract/post';
+
+export const postsRouter = express.Router();
+
+postsRouter.get('/posts/:post_id', guards.guest, handle(async ({ req }) => {
+	const post = await getPostByID(req.params.post_id);
+	if (!post || post.removed) {
+		throw new errors.notFound('Post not found');
+	}
```

Comment:
It just crossed my mind, moderators should be able to view removed posts. right?

This check will need a bypass for mods.